### PR TITLE
runtests: when <killserver> mentions http, kill http/2 too

### LINKED ISSUE #5289: Test 1700 failures
### I did this

Test 1700 to 1702 (HTTP/2) fail occasionally in my dev system when *not* run stand-alone (also visible in [some CI jobs](https://travis-ci.org/github/curl/curl/jobs/678911198#L5911)). I can reproduce pretty reliably if I run tests 350 to 1700. I'm in the process of trying to narrow down the list.

The failing builds probably have alt-svc enabled too.

### curl/libcurl version

current git master

### operating system

I've seen it on Linux but expect this to be platform independent.

### PR DESCRIPTION
**Pull request by bagder**
Since the http2 test server is a mere proxy that needs to know about the
dynamic port the HTTP server is using, it too needs to get restarted
when the http server is killed.

Fixes #5289 (regression caused by 80d6515).

### CODE REVIEW

File: tests/runtests.pl [hunk 1]
```diff
@@ -792,6 +793,12 @@ sub stopserver {
         # given a ssh server, also kill socks piggybacking one
         push @killservers, "socks${2}";
     }
+    if($server eq "http") {
+        # since the http2 server is a proxy that needs to know about the
+        # dynamic http port it too needs to get restarted when the http server
+        # is killed
+        push @killservers, "http/2";
```
Comment (emilengler) [inline tests/runtests.pl:800]: I'm not sure if that's the best approach. Maybe my suggestion is a bit too much for a such a simple task but I would create an associative array with `http` as key where the value is an array of all versions of that specific protocol. This approach is more and easier extensible.

Comment (bagder) [reply tests/runtests.pl:800]: We could of course change this concept and store some sort of generic dependency list per server instead, but that would mean a much larger change and I'm not seeing any obvious benefit with that.

Comment (emilengler) [reply tests/runtests.pl:800]: You're right, we can switch to that concept once we need to handle more protocol version servers

**Comment by emilengler** [comment]
Did the approve too early, clicked the button on accident. Will test now

**Comment by emilengler** [comment]
Ok, now the real approve of a4d510d0aac81f73a98dd635ace8ef5a1a1bdaaf.
Tested and works!

**Pull request closed without merging**